### PR TITLE
Field guide modal

### LINKED ISSUE #391: Add Field Guide
**Package**
lib-classifer

**Feature or Issue Description**
Field Guide store could be similar to how Tutorial is being handled. Store and UI needs to be added. Modal itself is in react-components

**To Dos**

- [ ] Example

### PR DESCRIPTION
**Pull request by srallen**
Package:

Closes #391.

Describe your changes:
Adds the UI for the field guide

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by srallen** [comment]
My project 1233 has a field guide. The default project loaded by the classifier does not.

**Comment by srallen** [comment]
I don't have this setup for the light/dark themes yet. This is an already large PR so I'll note in an issue.

**Comment by srallen** [comment]
This also has the issue of the size of the modal changing between the full list of items and selected item. I haven't come up with a better way to have it be responsive and have a max-width yet...

**Comment by eatyourgreens** [comment]
Testing this out in VoiceOver, I'm wondering if the item buttons should be links to the individual items? Give it a go and let me know what you think. When I press a button, it's not immediately obvious that I've navigated to the item, although I can continue navigating with VoiceOver and read the new text.

There's an annoying jump in height between an item loading and the linked image loading. Can we show a placeholder for images, or set a default height on that box.

File: packages/lib-classifier/src/store/FieldGuideStore.js [hunk 1]
```diff
@@ -92,11 +92,18 @@ const FieldGuideStore = types
...
     }
 
-    function setActiveItem (index) {
+    function setActiveItem (item) {
       const fieldGuide = self.active
-      if (fieldGuide && fieldGuide.items.length === index + 1 && fieldGuide.items[index]) {
-        self.activeItem = index
-        if (fieldGuide.items[index].icon) self.activeMedium = fieldGuide.items[index].icon
+      if (fieldGuide) {
+        const items = fieldGuide.items
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/FieldGuideStore.js]:
```suggestion
        const { items } = fieldGuide
```

File: packages/lib-classifier/src/store/FieldGuideStore.js [hunk 2]
```diff
@@ -92,11 +92,18 @@ const FieldGuideStore = types
...
+        const index = items.indexOf(item)
+        if (items[index]) {
+          const icon = items[index].icon
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/FieldGuideStore.js]:
```suggestion
          const { icon } = items[index]
```

File: packages/lib-classifier/src/store/FieldGuideStore.js [hunk 3]
```diff
@@ -92,11 +92,18 @@ const FieldGuideStore = types
...
+          if (icon) self.activeMedium = icon
+          return self.activeItem = index
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/FieldGuideStore.js]: Why is `self.activeItem` an index, and not the item? I find that confusing.

**Comment by srallen** [comment]
The new suggested changes feature is neat.

**Comment by eatyourgreens** [comment]
Yeah. I haven't tried it out for changes that are more than a few characters. Big changes might break things.

**Comment by srallen** [comment]
I updated to use an anchor instead and I like it a lot better for the hover state than what I came up with for the button.

**Comment by eatyourgreens** [comment]
VoiceOver says there's a missing counterpart translation in the link text somewhere.
Screenshot of the field guide item links, showing a missing translation error for the link text.

**Comment by srallen** [comment]
@eatyourgreens Oops! That's fixed in the latest commit.

**Comment by eatyourgreens** [comment]
I'm also getting `setActiveItem is not a function` when I click those links.

**Comment by srallen** [comment]
In addition to the refactor to use Anchor instead of Button for the items list, I've:
- Created an icon component that uses a placeholder fallback consistently between the list items view and the item details view
- Added Grommet's ResponsiveContext component to set the widths for the outer Box container. This fixes it between the item list view and item detail view and also makes it responsive. Note Enzyme does not yet support React's `createContext`, so I cannot test how `FieldGuide` works until Enzyme updates this. See: https://github.com/airbnb/enzyme/issues/1908
- Refactored how I was using Grid. It was just wrong before. I'm still learning!
- Finished the fix for renaming the `setActiveItemIndex` and `activeItemIndex`
- Fix the tests

File: packages/lib-classifier/src/store/FieldGuideStore.js [hunk 4]
```diff
@@ -92,19 +92,27 @@ const FieldGuideStore = types
       self.showModal = boolean
     }
 
-    function setActiveItem (index) {
+    function setActiveItemIndex (item) {
```
[the remark this answers is not in the document]
Comment (srallen) [reply packages/lib-classifier/src/store/FieldGuideStore.js]: The action is now using index consistently.

**Comment by srallen** [comment]
The focus bug I'll open as a separate issue. It might be a Grommet bug.

**Comment by eatyourgreens** [comment]
I've pulled down the rebased branch and the item links aren't working any more. I'll have a look to see if I can figure out why.

File: packages/lib-classifier/src/store/FieldGuideStore.js [hunk 5]
```diff
@@ -92,20 +92,14 @@ const FieldGuideStore = types
...
-        const { items } = fieldGuide
-        const index = items.indexOf(item)
-        if (items[index]) {
-          const { icon } = items[index]
-          if (icon) self.activeMedium = icon
-          return self.activeItemIndex = index
-        }
-
-        return self.activeItemIndex = undefined
+      if (fieldGuide && fieldGuide.items.length === index + 1 && fieldGuide.items[index]) {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/FieldGuideStore.js]: `fieldGuide.items.length === index + 1` is only true for the last item can be active at the moment.

Comment (eatyourgreens) [reply packages/lib-classifier/src/store/FieldGuideStore.js]: This should have been caught by a test too.

Comment (srallen) [reply packages/lib-classifier/src/store/FieldGuideStore.js]: I've fixed this and updated the tests too.

File: packages/lib-classifier/src/store/FieldGuideStore.spec.js [hunk 6]
```diff
@@ -32,7 +36,7 @@ const fieldGuideWithoutIcon = FieldGuideFactory.build({
 
 const project = ProjectFactory.build()
 
-describe('Model > FieldGuideStore', function () {
+describe.only('Model > FieldGuideStore', function () {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/FieldGuideStore.spec.js]: Stray `.only`

Comment (srallen) [reply packages/lib-classifier/src/store/FieldGuideStore.spec.js]: Whoops! That should be removed now.

**Comment by eatyourgreens** [comment]
Just waiting on Travis.

**Pull request merged**